lib/service/connect.go

```diff
@@ -67,8 +66,16 @@ func (process *TeleportProcess) reconnectToAuthService(role types.SystemRole) (*
 			// client works, by using call that should succeed at all times
 			if connector.Client != nil {
 				pingResponse, err := connector.Client.Ping(process.ExitContext())
-				if err := process.compareVersions(&pingResponse); err != nil {
-					process.log.Debugf("Error comparing formats: %v", err)
+
+				if !process.Config.SkipVersionCheck {
+					isOlder, compareErr := process.authServerTooOld(&pingResponse)
+					if compareErr != nil {
+						process.log.Errorf("Error comparing formats: %v", compareErr)
+						return nil, trace.Wrap(err)
+					}
+					if isOlder {
+						return nil, trace.NotImplemented("auth server is too old")
```

Comment:
nit: This error "auth server is too old" will be logged as an `ERROR` and will be the first thing the user sees rather than `WARN` "Only versions %d and greater are supported, auth server is version %d ...", the latter being much more descriptive.

Related to my other comment, we should aim for one good `WARN` log if `--skip-version-check=true` and one good `ERROR` log if `--skip-version-check=false`